src/plugins/ssh/index.ts

```diff
@@ -398,6 +394,6 @@ export const sshOrScp = async (args: {
     stdio: ["inherit", "inherit", "pipe"],
     debug: cmdArgs.debug,
     provider: request.type,
-    attemptsRemaining: sshProvider.maxRetries,
+    endTime: Date.now() + sshProvider.propagationTimeoutMs,
```

Comment:
To globally wait max propagationTimeoutMs, we would have to the endTime once before calling `preTestAccessPropagationIfNeeded` (above), and pass the same endTime to `spawnSshNode` (here).

Now for a GCP sudo access the total propagationTimeoutMs is 2m * 2.

Reply:
Done